packages/playwright-core/src/client/elementHandle.ts

```diff
@@ -270,6 +270,8 @@ export async function convertInputFiles(files: string | FilePayload | string[] |
     if (!items.every(item => typeof item === 'string'))
       throw new Error('File paths cannot be mixed with buffers');
     if (context._connection.isRemote()) {
+      if (!context._isLocalBrowser())
```

Comment:
I think server should also throw in this case, to avoid relying on clients 100%.